app/leaderboard/page.tsx

```diff
@@ -94,18 +96,32 @@ export default function Page() {
     };
 
     setLoading(true);
+    setRankingdataloading(true);
     fetchLeaderboardToppersResult({
       addr: requestBody.addr,
       duration: timeFrameMap(duration),
     });
     fetchRankingResults(requestBody);
     setLoading(false);
+    setRankingdataloading(false);
   }, [userAddress, status, apiCallDelay]);
 
   const fetchRankingResults = useCallback(
     async (requestBody: LeaderboardRankingParams) => {
-      const response = await fetchLeaderboardRankings(requestBody);
-      setRanking(response);
+      try {
+        setRankingdataloading(true);
```

Comment:
why do we set this to true here and also in line number - 99.
i think one doing it only once inside the function `fetchRankingResults` should work no?

Reply:
Yes it works but there's a lag between rendering already fetched data after loading state is false so I did it twice to ensure loading state component gets displayed